api/views/submission.py

```diff
@@ -121,6 +122,29 @@ def submit(self, request):
             raise ValidationError("{}".format(e))
 
         create_submission_action(submission)
+
+        if question.event.type == EVENT_TYPE_CHOICES.CHALLENGE:
+            course = question.course
+            course_reg = course.canvascourseregistration_set.filter(user=request.user)
+
+            event = question.event
+            team = event.team_set.filter(course_registrations__contains=[course_reg])
+
+            if team.course_registrations_set.count() > 1:
+                users = [course_reg.user for course_reg in team.course_registrations.all()]
+                solved_uqjs = UserQuestionJunction.objects.all().filter(user__in=users, is_solved=True)
+                solved_event_question_id = [
+                    solved_uqj.question.id
+                    for solved_uqj in solved_uqjs
+                    if solved_uqj.question.event.id is question.event.id
+                ]
+
+                solved_event_question_id = list(set(solved_event_question_id))
+                event_question_ids = [ele["id"] for ele in event.question_set.values_list("id")]
+
+                if len(event_question_ids) is len(solved_event_question_id):
+                    team_complete_challenge_action(question.event.id, team, request.user)
```

Comment:
This is a significant overhead for each submission. Also, it throws an error if a question is not in an event.
I think we can infer this data from what we currently store in this way:
You get all the submissions for an event. Then you check if all the questions have a correct submission and if yes then you get the last submission (based on time) and that is exactly what you are doing here.